toolchain/sem_ir/stringify_type.cpp

```diff
@@ -26,17 +26,22 @@ static auto GetTypePrecedence(InstKind kind) -> int {
 }
 
 namespace {
+// Contains the stack of steps for `StringifyTypeExpr`.
 class StepStack {
  public:
+  // The discriminator for the `Step` union.
   enum Kind : uint8_t {
     Inst,
     FixedString,
     ArrayBound,
     Name,
   };
+
+  // An individual step in the stack.
```

Comment:
This is all a bit abstract; I think it would help me to have more specifics about what a "step" represents. Maybe something like this?

```suggestion
  // An individual step in the stack, which prints some component of a type name.
```

(A more specific name like "PrintStep" might also help, but that may be out of scope here.)

Reply:
FWIW in the name direction, I think this is maybe okay right now because the file is just for stringification. But it might be worth it to rename `StepStack` to be `Stringifier`, refactoring `StringifyTypeExpr` to use a member function so that all the push/pop logic would be private.